widgetssdk/src/main/java/com/glia/widgets/view/head/controller/ActivityWatcherForChatHeadController.kt

```diff
@@ -23,30 +30,73 @@ internal class ActivityWatcherForChatHeadController(
 
     internal var screenSharingViewCallback: ScreenSharingController.ViewCallback? = null
 
+    override fun init() {
+        gliaOnEngagementUseCase.execute { watcher.addChatHeadLayoutIfAbsent() }
+        onEngagementEndUseCase.execute {
+            applicationChatHeadController.updateChatHeadView()
+        }
+        onCallVisualizerEndUseCase.execute(object : GliaOnCallVisualizerEndUseCase.Listener {
+            override fun engagementEnded() {
+                applicationChatHeadController.updateChatHeadView()
+            }
+        })
+    }
+
+    override fun shouldShowBubble(gliaOrRootView: String?): Boolean {
+        return shouldShowAppBubble(gliaOrRootView)
+    }
+
     override fun onActivityResumed(activity: Activity) {
         Logger.d(TAG, "onActivityResumed(root)")
-        serviceChatHeadController.onResume(watcher.fetchGliaOrRootView())
+        val gliaOrRootView = watcher.fetchGliaOrRootView()
+        serviceChatHeadController.onResume(gliaOrRootView)
+
+        val viewName: String = if (gliaOrRootView != null) gliaOrRootView::class.java.simpleName else ""
+        applicationChatHeadController.onResume(viewName)
+        if (Glia.isInitialized() && shouldShowAppBubble(viewName)) {
+            watcher.addChatHeadLayoutIfAbsent()
+        }
+
         setupScreenSharingViewCallback()
         screenSharingController.setViewCallback(screenSharingViewCallback)
     }
 
     override fun onActivityPaused() {
-        Logger.d(TAG, "onActivityPaused()")
-        serviceChatHeadController.onPause(watcher.fetchGliaOrRootView())
+        val gliaOrRootView = watcher.fetchGliaOrRootView()
+        serviceChatHeadController.onPause(gliaOrRootView)
+        val viewName: String = if (gliaOrRootView != null) gliaOrRootView::class.java.simpleName else ""
```

Comment:
just a small suggestion:
`val viewNmae: String = watcher.fetchGliaOrRootView()?.javaClass?.simpleName.orEmpty()`

Reply:
Cool, thank you 👍